cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -747,4 +768,17 @@ private void alterExploreStream(Id.Stream stream, boolean enable, @Nullable Form
   private void publishAudit(Id.Stream stream, AuditType auditType) {
     AuditPublishers.publishAudit(auditPublisher, stream, auditType, AuditPayload.EMPTY_PAYLOAD);
   }
+
+  private <T extends EntityId> void ensureAccess(T entityId, Action action) throws Exception {
+    ensureAccess(entityId, ImmutableSet.of(action));
+  }
+
+  private <T extends EntityId> void ensureAccess(T entityId, Set<Action> action) throws Exception {
+    Principal principal = authenticationContext.getPrincipal();
+    co.cask.cdap.api.Predicate<EntityId> filter = authorizer.createFilter(principal);
+    if (!Principal.SYSTEM.equals(principal) && !filter.apply(entityId)) {
+      throw new UnauthorizedException(principal, entityId);
+    }
+    authorizer.enforce(entityId, principal, action);
```

Comment:
This is a little confusing. If the purpose of this method is to enforce either one of the actions passed to it, then this statement should not be here. Instead, it should be on line 773, where you want to ensure a single action (or potentially all actions). Shouldn't use both filter and enforce.